nni/algorithms/compression/v2/pytorch/pruning/tools/base.py

```diff
@@ -375,10 +375,15 @@ def _expand_mask(self, name: str, mask: Tensor) -> Dict[str, Tensor]:
         weight_mask = mask.clone()
 
         if self.block_sparse_size is not None:
+            block_sparse_size = list(self.block_sparse_size)
+            # padding block_sparse_size to weight_mask length if dim is None
+            if self.dim is None:
+                for _ in range(len(weight_mask.size()) - len(self.block_sparse_size)):
+                    block_sparse_size.insert(0, 1)
```

Comment:
might be clearer:
head_part = [1 for _ in range(...)]
head_part.extend(block_sparse_size)